### PR TITLE
refactor: sfn handler

### PR DESCRIPTION
**Pull request by venjiang**
# Description

Refactor the signature of the sfn `Handler` to support multiple data writes.

## Impact

**Old**
```go
func Handler(data []byte) (frame.Tag, []byte) 
```
**Now**
```go
func Handler(ctx serverless.Context) 
```
**Methods** of `serverless.Context`  :
- `Tag() uint32` get incoming tag
-  `Data() []byte` get incoming data
-  `Write(tag uint32, data []byte) error` write data to zipper, support multiple writes

## Example
```go
func Handler(ctx serverless.Context) {
        // get ingress data tag
        tag := ctx.Tag()
        ...
	// get ingress data
	data := ctx.Data()
	// process data
	output := strings.ToUpper(string(data))
	outputTag := uint32(0x34)
	// send result to next processor
	ctx.Write(outputTag, []byte(output))
	output = []byte("write something")
	ctx.Write(outputTag, output)
	output = []byte("write others")
	ctx.Write(outputTag, output)
}
```

Close #532

### CODE REVIEW

File: cli/serverless/wasm/runtime.go [hunk 1]
```diff
@@ -3,15 +3,19 @@ package wasm
...
+	"github.com/yomorun/yomo/serverless"
 )
 
 // Define wasm import/export function names
 const (
+	WasmFuncStart          = "_start"
 	WasmFuncInit           = "yomo_init"
 	WasmFuncObserveDataTag = "yomo_observe_datatag"
-	WasmFuncLoadInput      = "yomo_load_input"
-	WasmFuncDumpOutput     = "yomo_dump_output"
```
Comment (wujunzhuo) [inline cli/serverless/wasm/runtime.go:13]: If these export functions were redefined, examples of other languages should also be modified, and docs as well.

File: serverless/context.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package serverless
+
+// Context sfn handler context
+type Context interface {
+	// Data incoming data
+	Data() []byte
+	// Tag incoming tag
+	Tag() uint32
+	// Write write data to zipper
+	Write(tag uint32, data []byte) error
```
Comment (wujunzhuo) [inline serverless/context.go:10]: How about defining a general `Writer` interface, which can be inherited by  both `Source` and `Context`?

File: serverless/guest/memory.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+var (
+	// ReadBuf is a buffer used to read data from the host.
+	ReadBuf = make([]byte, ReadBufSize)
+	// ReadBufPtr is a pointer to ReadBuf.
+	ReadBufPtr = uintptr(unsafe.Pointer(&ReadBuf[0]))
+	// ReadBufSize is the size of ReadBuf
+	ReadBufSize = uint32(2048)
+)
+
+func GetString(fn func(ptr uintptr, size uint32) (len uint32)) (result string) {
```
Comment (wujunzhuo) [inline serverless/guest/memory.go:17]: `GetString` could be implemented by wrapping `GetBytes`; besides, it seems to be unused?

Comment (venjiang) [reply serverless/guest/memory.go:17]: Yeah, it's not in use right now.

File: cli/serverless/wasm/runtime.go (hunk 1 above)
Comment (venjiang) [reply cli/serverless/wasm/runtime.go:13]: Yeah, we will introduce a new PR for examples and documents.

**Review by fanweixiao — APPROVED** [review]
👍🏻

**Review by woorui — APPROVED** [review]

**Pull request merged**